File: backend/src/main/java/com/rrs/entity/AiAgent.java
package com.rrs.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "ai_agent")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AiAgent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String code;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "agent_type", nullable = false, length = 50)
    private AgentType agentType;

    /** FK to Credential (LLM type) */
    @Column(name = "credential_id")
    private Long credentialId;

    @Column(name = "model_name", length = 200)
    private String modelName;

    @Column(name = "system_prompt", columnDefinition = "TEXT")
    private String systemPrompt;

    @Column(name = "temperature")
    private Double temperature;

    @Column(name = "max_tokens")
    private Integer maxTokens;

    @Column(name = "top_p")
    private Double topP;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "credential_id", insertable = false, updatable = false)
    private Credential credential;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/api/aiAgent.ts
import request from './request'

export interface AiAgentDTO {
  id: number
  code: string
  name: string
  description: string
  agentType: string
  credentialId: number | null
  credentialName: string | null
  modelName: string | null
  systemPrompt: string | null
  temperature: number | null
  maxTokens: number | null
  topP: number | null
  createdAt: string
  updatedAt: string
}

export interface AiAgentConfigDTO {
  credentialId?: number
  modelName?: string
  systemPrompt?: string
  temperature?: number
  maxTokens?: number
  topP?: number
}

export function getAiAgents(): Promise<AiAgentDTO[]> {
  return request.get('/api/ai-agents')
}

export function getAiAgent(id: number): Promise<AiAgentDTO> {
  return request.get(`/api/ai-agents/${id}`)
}

export function configureAiAgent(id: number, config: AiAgentConfigDTO): Promise<AiAgentDTO> {
  return request.put(`/api/ai-agents/${id}/config`, config)
}
